circuits/cpp/scripts/run_coverage

```diff
@@ -0,0 +1,78 @@
+#!bin/bash
+set -e
+
+# To be called LOCALLY for testing WITHOUT docker.
+#
+# Contains core logic for collecting test coverage in the current environment.
+#
+# Run from circuits/cpp/
+# Example:
+# ./scripts/run_coverage glob
+#
+# Running with the CLEAN environment variable set will delete the build-coverage folder and start again.
+# Running with the CLEAR_COV environment variable set will clear existing vscode coverage data (lcov.info).
+#
+# This will create two outputs: lcov.info and an html coverage report in build-coverage/coverage_report
+#
+# Note: Assumes ignition SRS transcripts have already been downloaded
+
+# TESTS:
+# 1. names of test executables to run
+#   - (like "aztec3_circuits_kernel_tests aztec3_circuits_abis_tests")
+# 2. or "glob" (LOWERCASE) to run all built tests
+# 3. or a file containing a list of test
+TESTS=$1
+
+# Set coverage build directory
+BUILD_DIR="build-coverage"
+
+# If the clean file is set, wipe all existing coverage data
+if [[ -n "$CLEAN" ]]; then
+  echo "The CLEAN environment variable is set. Clearing existing coverage data."
+  rm -rf ./$BUILD_DIR || true
+fi
+
+# Perform coverage build
+echo "Setting cmake to run with preset coverage"
+cmake -Bbuild-coverage -DCMAKE_BUILD_TYPE=Debug -DUSE_TURBO=true --preset=coverage .
+
+# if TESTS is GLOB or empty, run all built tests
+# if TESTS is a file, assume it contains a list of tests to run
+# otherwise assume TESTS itself is a list of tests to run
+if [ "$TESTS" == "glob" ] || [ -z "$TESTS" ]; then
+  echo "    No test file specified. Building all test files."
+  TESTS=$(cat ./scripts/a3-tests)
+elif [ -f "$TESTS" ]; then
+  TESTS=$(cat $TESTS | tr '\n' ' ')
+fi
+
+cd $BUILD_DIR
+
+# if just the CLEAR_COV env var is set then we will clear existing coveage report information
+if [[ -n "$CLEAR_COV" ]]; then
+  echo "The CLEAR_COV environment variable is set. Clearing existing coverage data."
+  rm ./lcov.info || true
+  rm ./merged_profdata/default.profdata || true
+  rm ./default.profraw || true
+  rm -rf ./bin/profdata || true
+fi
+
+for BIN in $TESTS; do
+  echo "*** Building $BIN ***"
+  cmake --build . --target $BIN
```

Comment:
should this be `cmake --build --preset coverage --target $BIN` instead? Shouldn't need to specify the build dir if using the preset.

Reply:
Might need to `cd ..` to use the preset, but I still think it's maybe the better option since the preset may include additional settings that we want (in the future at least). To do the `cd ..` in a subshell so that it doesn't actually change current dir of the following lines you could do `(cd .. && cmake --build --preset coverage --target $BIN)` (parens create subshell).

I don't feel strongly about this so it's probably fine to leave it as-is. Just trying to think about the "right" way to handle this.